### PR TITLE
Allow host list cache

### PR DESCRIPTION
**Pull request by jacogr**
Closes https://github.com/polkadot-js/phishing/issues/24

cc @Tbaut - would just like a sanity check

### CODE REVIEW

File: packages/phishing/src/index.ts [hunk 1]
```diff
@@ -7,6 +7,10 @@ import fetch from '@polkadot/x-fetch';
 
 // Equivalent to https://raw.githubusercontent.com/polkadot-js/phishing/master/all.json
 const ALL_JSON = 'https://polkadot.js.org/phishing/all.json';
+const CACHE_TIMEOUT = 1 * 60 * 60 * 1000;
```
Comment (jacogr) [inline packages/phishing/src/index.ts:10]:
```suggestion
const CACHE_TIMEOUT = 1 * 24 * 60 * 1000;
```

**Comment by Tbaut** [comment]
Looks good from the bare eyes, I'll build it in the morning and test it live.

File: packages/phishing/src/index.ts [hunk 2]
```diff
@@ -18,10 +22,19 @@ function extractHost (path: string): string {
 /**
  * Retrieve allow/deny from our list provider
  */
-export async function retrieveHostList (): Promise<HostList> {
+export async function retrieveHostList (allowCached = true): Promise<HostList> {
+  const now = Date.now();
+
+  if (allowCached && cacheList && now < cacheEnd) {
```
Comment (jacogr) [inline packages/phishing/src/index.ts:28]:
```suggestion
  if (allowCached && cacheList && (now < cacheEnd)) {
```

File: packages/phishing/src/index.ts (hunk 1 above)
Comment (jacogr) [inline packages/phishing/src/index.ts:10]:
```suggestion
// 1 hour cache refresh
const CACHE_TIMEOUT = 1 * 60 * 60 * 1000;
```

**Pull request merged**